deepchecks/tabular/checks/train_test_validation/category_mismatch_train_test.py

```diff
@@ -92,40 +107,41 @@ def run_logic(self, context: Context) -> CheckResult:
                 new_category_counts = dict(test_column.value_counts()[new_category_values])
                 new_categories_ratio = sum(new_category_counts.values()) / n_test_samples
                 sorted_new_categories = dict(sorted(new_category_counts.items(), key=lambda x: x[1], reverse=True))
-                new_categories[feature] = sorted_new_categories
-                if context.with_display:
-                    display_data.append([feature, len(new_category_values), new_categories_ratio,
-                                        list(sorted_new_categories.keys())[:self.max_new_categories_to_show]])
+                result_data.append([feature, len(new_category_values), new_categories_ratio,
+                                    list(sorted_new_categories.keys()), feature_importance[feature]])
             else:
-                new_categories[feature] = {}
+                result_data.append([feature, 0, 0, [], feature_importance[feature]])
 
-        # Display
-        if display_data:
-            display = pd.DataFrame(data=display_data,
+        result_data = pd.DataFrame(data=result_data,
                                    columns=['Column',
                                             'Number of new categories',
                                             'Percent of new categories in sample',
-                                            'New categories examples'])\
-                                    .set_index(['Column'])
+                                            'New categories in column',
+                                            'Feature importance']).set_index(['Column'])
+        result_data.sort_values(by='Percent of new categories in sample')
+        if any(feature_importance.isna()):
```

Comment:
Could there be a situation when only certain members of feature_importance are na? because in that case we may want to keep it. 
If so, maybe change to `all`?

Reply:
Sure